modules/tfm/tfm/boards/common/nrf_provisioning.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "tfm_plat_provisioning.h"
+#include "tfm_plat_otp.h"
+#include "tfm_hal_platform.h"
+#include "tfm_attest_hal.h"
+#include "hw_unique_key.h"
+#include "nrfx_nvmc.h"
+#include <nrf_cc3xx_platform.h>
+#include <nrf_cc3xx_platform_identity_key.h>
+
+int tfm_plat_provisioning_is_required(void)
+{
+    enum tfm_security_lifecycle_t lcs = tfm_attest_hal_get_security_lifecycle();
+
+    return lcs != TFM_SLC_SECURED;
```

Comment:
It would be more safe and future proof to list the states that are accepted to do provisioning in.

This code would try to provision in decommisioned state for example.

Also the code the perform function returns error code unless you are in "TFM_SLC_PSA_ROT_PROVISIONING", so returning lcs == TFM_SLC_PSA_ROT_PROVISIONING seems to be the correct one.
```suggestion
    return lcs == TFM_SLC_PSA_ROT_PROVISIONING;
```